components/nexus-bootstrap/src/main/java/org/sonatype/nexus/bootstrap/jetty/ConnectorManager.java

```diff
@@ -0,0 +1,262 @@
+/*
+ * Sonatype Nexus (TM) Open Source Version
+ * Copyright (c) 2008-present Sonatype, Inc.
+ * All rights reserved. Includes the third-party code listed at http://links.sonatype.com/products/nexus/oss/attributions.
+ *
+ * This program and the accompanying materials are made available under the terms of the Eclipse Public License Version 1.0,
+ * which accompanies this distribution and is available at http://www.eclipse.org/legal/epl-v10.html.
+ *
+ * Sonatype Nexus (TM) Professional Version is available from Sonatype, Inc. "Sonatype" and "Sonatype Nexus" are trademarks
+ * of Sonatype, Inc. Apache Maven is a trademark of the Apache Software Foundation. M2eclipse is a trademark of the
+ * Eclipse Foundation. All other trademarks are the property of their respective owners.
+ */
+package org.sonatype.nexus.bootstrap.jetty;
+
+import java.util.ArrayList;
+import java.util.IdentityHashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.eclipse.jetty.http.HttpScheme;
+import org.eclipse.jetty.server.HttpConfiguration;
+import org.eclipse.jetty.server.HttpConnectionFactory;
+import org.eclipse.jetty.server.Server;
+import org.eclipse.jetty.server.ServerConnector;
+import org.eclipse.jetty.server.SslConnectionFactory;
+import org.eclipse.jetty.util.ssl.SslContextFactory;
+
+import static java.util.Collections.unmodifiableList;
+
+/**
+ * Jetty's internal connector manager to manage custom {@link ConnectorConfiguration}s.
+ * <p/>
+ * Note: This class relies on "id" properties in Jetty XML configuration! Hence, the existing beans in Jetty XML
+ * should NOT be renamed, while adding new (with unique, non-clashing) IDs is okay thing to do.
+ */
+public class ConnectorManager
+{
+  private static final String HTTP_CONFIG_ID = "httpConfig";
+
+  private static final String HTTP_CONNECTOR_ID = "httpConnector";
+
+  private static final String SSL_CONTEXT_FACTORY_ID = "sslContextFactory";
+
+  private static final String HTTPS_CONFIG_ID = "httpsConfig";
+
+  private static final String HTTPS_CONNECTOR_ID = "httpsConnector";
+
+  private final Server server;
+
+  private final Map<String, Object> namedBeans;
+
+  private final IdentityHashMap<ConnectorConfiguration, ServerConnector> managedConnectors;
+
+  private final List<ConnectorConfiguration> defaultConnectors;
+
+  public ConnectorManager(final Server server,
+                          final Map<String, Object> namedBeans)
+  {
+    this.server = server;
+    this.namedBeans = namedBeans;
+    this.managedConnectors = new IdentityHashMap<>();
+    this.defaultConnectors = unmodifiableList(buildDefaultConnectors());
+  }
+
+  public List<ConnectorConfiguration> defaultConnectors() {
+    return defaultConnectors;
+  }
+
+  /**
+   * Adds and starts connector to Jetty based on passed in configuration.
+   */
+  public ServerConnector addConnector(final ConnectorConfiguration connectorConfiguration)
```

Comment:
Minor, but this method is doing a lot - serving as a registry of supported schemes, defining factories for various server connectors, populating shared fields, then registering, adding, and starting the connector.